metrics.md

```diff
@@ -30,6 +30,55 @@ The following are the minimal metrics agreed to be conformed by the various clie
+| `beacon_data_availability_reconstruction_time_seconds`            | Histogram | Time taken to reconstruct columns                                      | On data column kzg verification  |
+| `beacon_data_column_sidecar_computation_seconds`                  | Histogram | Time taken to compute data column sidecar, including cells, proofs and inclusion proof                |  On data column sidecar computation            |
+| `beacon_data_column_sidecar_inclusion_proof_verification_seconds` | Histogram | Time taken to verify data column sidecar inclusion proof                          |  On data column sidecar inclusion proof verification  |
+| `beacon_kzg_verification_data_column_single_seconds`              | Histogram | Runtime of single data column kzg verification                                 | On single data column kzg verification  |
+| `beacon_kzg_verification_data_column_batch_seconds`               | Histogram | Runtime of batched data column kzg verification                                 | On batched data column kzg verification |
+| `beacon_custody_columns_count_total`                              | Counter     | Total count of columns in custody within the data availability boundary                                     | On custody collecting and verification |
+
+#### Gossipsub metrics
+Gossipsub domain metrics should contain topic `data_column_sidecar_{subnet_id}` as labels.
+
+| Name | Metric type | Usage | Sample collection event |
+|--------------------------------------------|-------------|-------------------------------------------------------------|----------------------|
+| `gossipsub_topic_msg_sent_counts_total` | Counter | Number of gossip messages sent to each topic | On sending a message over a topic |
+| `gossipsub_topic_msg_sent_bytes_total` | Counter | Number of bytes sent to each topic | On sending a message over a topic |
+| `gossipsub_topic_msg_recv_counts_unfiltered_total` | Counter | Number of gossip messages received from each topic (including duplicates)  | On receiving a message from a topic including duplicates|
+| `gossipsub_topic_msg_recv_bytes_unfiltered_total` | Counter | Number of bytes received from each topic (including duplicates) |  On receiving a message from a topic including duplicates |
+| `gossipsub_topic_msg_recv_counts_total` | Counter | Number of gossip messages received from each topic (deduplicated)  | On receiving a message from a topic deduplicated |
+| `gossipsub_topic_msg_recv_bytes_total` | Counter | Number of bytes received from each topic (deduplicated) | On receiving a message from a topic deduplicated |
+
+
+#### Req/Resp metrics 
+
+Req/Resp domain metrics should contain protocol ID `/eth2/beacon_chain/req/data_column_sidecars_by_root/1/` and `/eth2/beacon_chain/req/data_column_sidecars_by_range/1/` as labels.
```

Comment:
Should we standardize the label name to `protocolId` or something similar?  Only issue I see with that would be naming conventions across languages (ie JS is `procolId` and rust would be `protocol_id`).  But would allow a common grafana dashboard to work for all clients.

Same question for `topic` above in the gossipsub section

Reply:
@matthewkeil Yup standardizing the label names makes sense - [they should be in snake case](https://prometheus.io/docs/instrumenting/writing_exporters/#:~:text=Prometheus%20metrics%20and%20label%20names,to%20leave%20them%20as%2Dis.)

> Prometheus metrics and label names are written in snake_case